### PR TITLE
rebase onto latest upstream

### PR DESCRIPTION
**Pull request by sanchezl**
Rebase onto latest upstream.

### CODE REVIEW

File: pkg/initializer/discover_test.go [hunk 1]
```diff
@@ -355,7 +355,7 @@ func TestFindMigratableResources(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	expected := []schema.GroupVersionResource{schema.GroupVersionResource{Group: "extensions", Version: "v1beta1", Resource: "daemonsets"}}
+	expected := []schema.GroupVersionResource{{Group: "extensions", Version: "v1beta1", Resource: "daemonsets"}}
```
Comment (deads2k) [inline pkg/initializer/discover_test.go]: why?

File: cmd/migrator/app/migrator.go [hunk 2]
```diff
@@ -24,8 +23,8 @@ const (
 
 var (
 	kubeconfigPath = flag.String("kubeconfig", "", "absolute path to the kubeconfig file specifying the apiserver instance. If unspecified, fallback to in-cluster configuration")
-	kubeAPIQPS     = flag.Float32("kube-api-qps", rest.DefaultQPS, "QPS to use while talking with kubernetes apiserver.")
-	kubeAPIBurst   = flag.Int("kube-api-burst", rest.DefaultBurst, "Burst to use while talking with kubernetes apiserver.")
+	kubeAPIQPS     = flag.Float32("kube-api-qps", 40.0, "QPS to use while talking with kubernetes apiserver.")
```
Comment (deads2k) [inline cmd/migrator/app/migrator.go:26]: @sanchezl get this fixed upstream

**Comment by deads2k** [comment]
/lgtm
/hold

hold to be sure I only need to review "UPSTREAM" commits.  @sanchezl please confirm

File: cmd/migrator/app/migrator.go (hunk 2 above)
Comment (sanchezl) [reply cmd/migrator/app/migrator.go:26]: @deads2k I was planning on adding the `--kube-api-qps` and `"kube-api-burst` options on our invocation of the migrator and then just skipping this patch on the next rebase.

**Comment by sanchezl** [comment]
> hold to be sure I only need to review "UPSTREAM" commits. @sanchezl please confirm

@deads2k only UPSTREAM commits need review.

**Comment by deads2k** [comment]
/hold cancel
Well, the next kube bump promises to be interesting.

**Pull request merged**